link/link_test.go

```diff
@@ -48,13 +49,43 @@ func TestRawLink(t *testing.T) {
 
 	testLink(t, link, testLinkOptions{
 		prog: prog,
-		loadPinned: func(f string) (Link, error) {
-			return LoadPinnedRawLink(f)
+		loadPinned: func(f string, opts *ebpf.LoadPinOptions) (Link, error) {
+			return LoadPinnedRawLink(f, UnspecifiedType, opts)
 		},
 	})
 }
 
-func mustCgroupFixtures(t *testing.T) (*os.File, *ebpf.Program, func()) {
+func TestRawLinkLoadPinnedWithOptions(t *testing.T) {
+	cgroup, prog := mustCgroupFixtures(t)
+
+	link, err := AttachRawLink(RawLinkOptions{
+		Target:  int(cgroup.Fd()),
+		Program: prog,
+		Attach:  ebpf.AttachCGroupInetEgress,
+	})
+	testutils.SkipIfNotSupported(t, err)
+	if err != nil {
+		t.Fatal("Can't create raw link:", err)
+	}
+
+	path := filepath.Join(testutils.TempBPFFS(t), "link")
+	err = link.Pin(path)
+	testutils.SkipIfNotSupported(t, err)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// It seems like the kernel ignores BPF_F_RDONLY when updating a link,
```

Comment:
Should we add this mote to the public API as well?

Reply:
I hope that https://lore.kernel.org/bpf/20210326160501.46234-1-lmb@cloudflare.com/ gets somewhere, which would kind of make this obsolete. Keeping documentation up with kernel bugs is a lost cause I think.